src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
After you set the config file, this function also searches for more config files in parent directories, etc., which shouldn't happen when you pass a config file via `--config`. Instead of doing this here, I think you need to refactor this function into several small ones so you can just get the functionality you want (such as merging configuration).